common/src/main/java/org/apache/drill/common/types/Types.java

```diff
@@ -27,10 +27,14 @@
 import org.apache.drill.common.types.TypeProtos.MinorType;
 
 import com.google.protobuf.TextFormat;
+import org.apache.drill.common.util.CoreDecimalUtility;
 
 public class Types {
   static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(Types.class);
 
+  public static final int MAX_VARCHAR_LENGTH = 65536;
```

Comment:
Oracle seems to allow up to 4000 bytes for varchar [1].  The number Drill allows seems to be bigger than 4k.

This magic number probably comes from Calcite, if I understand correctly.  In RDBMS, if a string goes beyond the limit, people is recommended to use LOB/BLOB in stead. 

1. https://docs.oracle.com/cd/B28359_01/server.111/b28320/limits001.htm#i287903